### DISCUSSION TITLE
Embedding H2o Wave App into other app frameworks?

### DISCUSSION THREAD

**Discussion by audreyleteve** [h2oai/wave#2388] [Q&A] [answered]
Hello Team!

Is there a supported way to embed a h2o wave app into another application? eg. via API embedding or iframe embedding
We are hoping to create wave "templates" that are relatively agnostic to whatever tech stack is in use. Something like an iframe embed, so that we could create small wave apps and utilize these apps on existing websites. 

thank you.

**Comment by mturoci** [comment, marked as answer]
Hello Audrey. There are 2 different kinds of embeddings:

1. Iframe - what you want probably. Iframes are simply "windows to other sites within a website", e.g. you can embed a youtube video in an iframe on your website. For that, you can use existing Wave, but the app needs to be hosted publicly and the embedding website will just need an iframe with a URL pointing to your app.
2. Embed within an existing framework - Lightwave. Say you have an existing Flask, Django, FastAPI app and want to add a UI onto the same server and application code to leverage existing codebase, auth, and whatnot. This is where Lightwave comes into play as it is just a library without its own server, serving only a single very specific use case - paint the UI (unlike Wave, which handles files, authentication, etc.)
